### PR TITLE
Xeno Ability Cleanup and Fixes

### PR DESCRIPTION
**Pull request by Drulikar**
## About The Pull Request
This PR is a collection of QoL changes, polish, and fixes for various xeno ability buttons and their associated hotkeys. It largely focuses on fixing inconsistencies with ability button arrangements, queen administrative abilities not using plasma, ability icons not updating their status (choose resin, gut, and grow ovipositor), and ability primacy values (the abilities that are designated to utilize Xeno-Primary-Action-One, Xeno-Primary-Action-Two, etc.) utilizing the new XENO_PRIMARY_ACTION_5 from #694. I also added a bindable hotkey for Queen Screech, made several abilities instant use (rather than MMB/Shift click), allowed all builder caste xenos (minus gardener) to use Order Construction as XENO_PRIMARY_ACTION_5, added Generate Egg as XENO_PRIMARY_ACTION_3 for the Eggcarrier strain, and set several toggle abilities to use the purple border button template when they are active. See change log below for all changes.

## Why It's Good For The Game
Allows Queens to actually use the new hotkey system for their most commonly used abilities, button arrangement consistency, general quality of life for abilities, and fixes. Previously mature mobile Queens couldn't even select their spit ability via hotkey (only change its type).

New button arrangement as young mobile Queen:
image

New button arrangement as an ovi Queen:
image

New button arrangement as a mature mobile Queen:
image

Better indicated cooldowns for Queen abilities (button arrangement changed slightly since this image was taken):
image

New button arrangement as a Hivelord (just puts transfer plasma in same spot as a drone):
image

New button arrangement as a Warden (just swaps protect and retrieve):
image

New button arrangement as a Eggsac Carrier (just puts retrieve egg at the end again):
image

More purple button template for toggleable/duration abilities:
image

## Changelog

:cl: Drathek
add: Added Screech macro hotkey for Queen Screech. YOU SHOULD RE-BIND SCREECH AS QUEEN, IT IS NOT IN THE DEFAULT XENO-ACTION MACRO LIST.
qol: Changed mobile Queen ability primary hotkeys to 1:Weed, 2:Change Spit, 3:Spit, 4:Choose Resin Structure, 5:Secrete Resin.
qol: Changed ovi Queen ability primary hotkeys to 1:Heal, 2:Plasma, 3:Expand Weeds, 4:Choose Resin Structure, 5:Secrete Resin.
qol: Set Order Construction as fifth primary ability hotkey for any caste that can use it except Queen and Gardener.
qol: Set Generate Egg as third primary ability hotkey for Eggsac Carrier strain. (1:Weed, 3:Generate Egg, 4:Retrieve Egg, 5:Order Construction).
qol: Set Change Fruit as fifth primary ability hotkey for Gardener strain. (1:Weed, 2:Hardy Weed, 3:Resin Surge, 4:Plant Fruit, 5:Change Fruit)
qol: Shifted primary ability hotkeys for Acid Runner to 1:Acid, 2:For the Hive.
qol: Added 10s cooldown to Word of the Queen ability and implemented button status for the cooldown.
qol: Added cooldown status for Queen's Gut ability.
qol: Added cooldown status for Grow Ovipositor ability.
qol: Rearranged ability buttons for Queen, base/whisperer Hivelords, Eggsac Carrier, and Warden.
qol: The following abilities now are instantly cast instead of select and then MMB/Shift click: Screech (Queen), Empower (Rav), Spike Shield (Hedgehog), Spike Shed (Hedgehog), Dodge (Dancer), Tremor (Burrower), Roar (Abom), and Smash (Abom).
qol: Eviscerate (Rav) is now instantly cast when used via hotkey, but not when selected by mouse (this is the existing behavior for Defender Fortify).
qol: The following abilities now use the purple button template to indicate active state (like Use Pain already does): Hide, Rest, Toggle Long Range Sight, Resin Walker (Hivelords), Toggle Charging (Charger), Toggle Crest Defense (Defenders), Fortify (Defenders), Empower (Rav), Spike Shield (Hedgehog), Dodge (Dancer), Paralyzing Slash (Sentinel), Crippling Strike (Lurker), Turn Invisible (Lurker), and Defensive Shield (Crusher).
qol: Added a 0.5 second spam prevention measure when toggling the same ability off to help prevent accidental de-selecting.
fix: Fixed Corrosive Acid incorrectly being detected as XENO_PRIMARY_ACTION_5 due to overlap in its define with XENO_CORROSIVE_ACID (it has its own dedicated macro anyways).
fix: Fixed Word of the Queen ability button missing for pre-ovi young queen.
fix: Fixed some abilities such as Remove Eggsac ability re-arranging after second ovi.
fix: Fixed Choose Resin Structure ability icon not updating when changing ovi state or becoming a resin whisperer (it basically always reverts back to resin wall on change).
fix: Fixed Queen's Gut ability preventing new hive orders when it was on cooldown.
fix: Fixed possibility of using plasma for hive orders when unconscious.
fix: Fixed possibility of using plasma for Word of the Queen when canceling the message.
fix: Fixed De-evolve ability not using plasma nor actually requiring a reason.
fix: Fixed Banish ability not using plasma nor actually requiring a reason.
fix: Fixed Re-admit ability not using plasma.
fix: Fixed statistics tracking when using De-evolve ability: It was still tracking old mob and incorrectly incrementing the participant count.
fix: Fixed missing ability_name for Charger Tumble.
fix: Fixed incorrect ability_name for Shaman Remember Pain.
fix: Fixed selecting xeno actions via verb when hidden (should only apply to queen).
code: Improved more variable readability.
code: Removed unused De-evolve ability code in Abilities.dm.
code: Added /datum/action/xeno_action/activable/proc/action_deselect to be an overridable proc called when a different action is clicked on that causes the current action to be deselected.
/:cl:

### CODE REVIEW

**Comment by stanalbatross** [comment]
> BTW: Do we like the queen screech button being a toggle when its otherwise coded to be instant (both the verb and new hotkey perform handle_xeno_macro_datum which instantly activates it because it is a XENO_ACTION_ACTIVATE)?

instant better

**Comment by Diegoflores31** [comment]
Drukilar now that you are touching the queen verbs any chance that you can add an actually cooldown indicator for the GIB ? it should just a 1 line check because as it is rn it doesnt show how much cooldown you have left and it just shows the ability as available regardless

**Comment by stanalbatross** [comment]
> I have nothing else I can think of needing to change unless there is feedback. The only real reason why the line changes are so large is just because so much code does not adhere to code style standards.

ok review time

**Review by stanalbatross — CHANGES_REQUESTED** [review]
- action five as gardener should be select fruit
- no primary action 1 for acid runner? move all the macros down one
- issue with crusher tumble, has a null name, not sure if related
image
- you can change warden order if you want, nanu kinda merged the pr before I could change it
-

**Comment by usnpeepoo** [comment]
berserker eviscerate no longer being selectable isn't a good idea, many berserkers select it early/before they hit max rage to use it at a good time/decide not to and making it one click and done would make handling the ability a lot harder. Plus the root in place aspect makes a misclick deadly (esp. at lower levels if you do misclick since the windup is high as hell). Love everything else about this PR, great qol

File: code/modules/mob/living/carbon/xenomorph/abilities/xeno_action.dm [hunk 1]
```diff
@@ -135,6 +135,8 @@
 /datum/action/xeno_action/activable/action_activate()
 	if(!owner)
 		return
+	if(src.hidden)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/xeno_action.dm:138]: remove src

**Review by stanalbatross — APPROVED** [review]
otherwise fine

**Pull request merged**